numba/tests/test_remove_dead.py

```diff
@@ -3,17 +3,32 @@
 # SPDX-License-Identifier: BSD-2-Clause
 #
 
+import numba
 from numba import compiler, typing
+from numba.compiler import compile_isolated, Flags
 from numba.targets import cpu
 from numba import types
 from numba.targets.registry import cpu_target
 from numba import config
 from numba.annotations import type_annotations
-from numba.ir_utils import copy_propagate, apply_copy_propagate, get_name_var_table, remove_dels, remove_dead
+from numba.ir_utils import (copy_propagate, apply_copy_propagate,
+                            get_name_var_table, remove_dels, remove_dead,
+                            remove_call_handlers)
 from numba import ir
 from numba import unittest_support as unittest
 import numpy as np
 
+# use xxnrm2 to test call a C function with ctypes
+from numba.targets.linalg import _BLAS
+xxnrm2 = _BLAS().numba_xxnrm2(types.float64)
+
+def remove_dead_xxnrm2(rhs, lives, call_list):
+    if call_list == [xxnrm2]:
+        return rhs.args[4].name not in lives
+    return False
+
+remove_call_handlers.append(remove_dead_xxnrm2)
```

Comment:
Is this applying a permanent state mutation that will persist across the test suite? If so it ought to be context managed. It also seems like it is only used in the `test_alias_ctypes` so it might well be easier to just apply the change locally in that test?

Reply:
done.